ow_lander/src/ow_lander/faults.py

```diff
@@ -0,0 +1,145 @@
+
+  def __init__(self):
+    self._status = [
+      GoalStatus() for _ in range(ActionGoalStatus.NUM_GOAL_TYPES)
+    ]
+    self._goal_state_pub = rospy.Publisher(
+      self.ACTION_GOAL_STATUS_TOPIC, ActionGoalStatus,
+      queue_size = 1, latch = True
+    )
+
+  def publish(self, group_id, status):
+    if group_id is None:
+      return
+    # update internal status list
+    timestamp = rospy.Time.now()
+    goal_id = str(group_id) # ActionGoalStatus emum value to string
+    self._status[group_id].goal_id.stamp = timestamp
+    self._status[group_id].goal_id.id    = goal_id
+    self._status[group_id].status        = status
+    # publish status
+    msg = ActionGoalStatus()
+    msg.header.stamp    = timestamp
+    msg.header.frame_id = 'world'
+    msg.status_list     = self._status
+    self._goal_state_pub.publish(msg)
+
+  def get_group_status(self, group_id):
+    return self._status[group_id].status
+
+  def get_group_str(self, group_id):
+    if group_id == ActionGoalStatus.ARM_GOAL:
+      return "ARM"
+    elif group_id == ActionGoalStatus.TASK_GOAL:
+      return "TASK"
+    if group_id == ActionGoalStatus.CAMERA_GOAL:
+      return "CAMERA"
+    if group_id == ActionGoalStatus.PAN_TILT_GOAL:
+      return "PAN_TILT"
+    else:
+      return "UNKNOWN GROUP"
+
+
+class ArmFaultsPublisher(metaclass = Singleton):
+
+  INTERNAL_ARM_FAULT_TOPIC = "arm_faults_internal"
+
+  def __init__(self):
+    self._flags = ArmFaultsStatus.NONE
+    self._arm_faults_internal_pub = rospy.Publisher(
+      self.INTERNAL_ARM_FAULT_TOPIC, ArmFaultsStatus,
+      queue_size = 1, latch = True
+    )
+
+  def publish(self, flags):
+    self._arm_faults_internal_pub.publish(value = flags)
+
+
+class SystemFaultsInterface:
+
+  def __init__(self, group_id):
+    self._goal_fault_publisher = SystemFaultsPublisher()
+    self._group_id = group_id
+
+  def reset_system_faults(self):
+    # HACK: The SUCCEEDED state does not reflect the current state of the action
+    #   that has been called. It cannot have succeeded yet because it has yet to
+    #   be attempted here. This is a hack. ActionGoalStatus.msg is effectively
+    #   interpreted by ow_fault_detector as a boolean array because an element
+    #   can either be SUCCEEDED or ABORTED. To reset system_faults_status goal
+    #   error flags to 0 at the beginning of an action, we broadcast SUCCEEDED.
+    self._notify(GoalStatus.SUCCEEDED)
+
+  def notify_succeeded(self):
+    self._notify(GoalStatus.SUCCEEDED)
+
+  def notify_aborted(self):
+    self._notify(GoalStatus.ABORTED)
+
+  def notify_preempted(self):
+    self._notify(GoalStatus.PREEMPTED)
+
+  def is_faulted(self):
+    return self._goal_fault_publisher.get_group_status(
+      self._group_id) == GoalStatus.ABORTED
+
+  def get_rejected_message(self):
+    group = self._goal_fault_publisher.get_group_str(self._group_id)
+    return f"Action group, {group}, is faulted. Use the  FaultClear action " \
```

Comment:
```suggestion
    return f"Action group, {group}, is faulted. Use the FaultClear action " \
```
Will that better leave only one space between "the" and "Faultclear"?

Reply:
Thank you!